Calling virtual method from CanDriver destructor is invalid

`clang-tidy` spots this.

```
include/CanDriver.hpp:79:36: warning: Call to virtual method 'CanDriver::uninitialiseSocketCan' during destruction bypasses virtual dispatch [clang-analyzer-optin.cplusplus.VirtualCall]
   79 |             virtual ~CanDriver() { uninitialiseSocketCan(); } //!< Destructor
```

basically, only `unitialiseSocketCan()` in the base `CanDriver` will be called. See [cppreference.com](https://en.cppreference.com/w/cpp/language/virtual.html#During_construction_and_destruction). The same remarks will apply to `initialiseSocketCan()`.

I'd suggest making them both non-virtual private methods and updating the README.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.23)
 
-project(sockcanpp LANGUAGES CXX VERSION 1.7.1 DESCRIPTION "A simple C++ wrapper for the Linux SocketCAN API" HOMEPAGE_URL "https://github.com/simoncahill/libsockcanpp")
+project(sockcanpp LANGUAGES CXX VERSION 1.7.2 DESCRIPTION "A simple C++ wrapper for the Linux SocketCAN API" HOMEPAGE_URL "https://github.com/simoncahill/libsockcanpp")
 
 option(BUILD_SHARED_LIBS "Build shared libraries (.so) instead of static ones (.a)" ON)
 option(BUILD_TESTS "Build the tests" OFF)

--- a/include/CanDriver.hpp
+++ b/include/CanDriver.hpp
@@ -155,9 +155,9 @@ namespace sockcanpp {
             virtual void                setReceiveOwnMessages(const bool enabled = true) const; //!< Sets the receive own messages option for the interface
             virtual void                setReturnRelativeTimestamps(const bool enabled = true) { m_relativeTimestamps = enabled; }
 
-        protected: // +++ Socket Management +++
-            virtual void                initialiseSocketCan(); //!< Initialises socketcan
-            virtual void                uninitialiseSocketCan(); //!< Uninitialises socketcan
+        private: // +++ Socket Management +++
+            void                        initialiseSocketCan(); //!< Initialises socketcan
+            void                        uninitialiseSocketCan(); //!< Uninitialises socketcan
 
         private: // +++ Member Functions +++
             virtual CanMessage          readMessageLock(bool const lock = true); //!< readMessage deadlock guard

